library/src/G4LatticeReader.cc

```diff
@@ -110,9 +110,23 @@ G4bool G4LatticeReader::OpenFile(const G4String& filename) {
 
   G4String filepath = filename;
   psLatfile = new std::ifstream(filepath);
-  if (!psLatfile->good()) {			// Local file not found
+  if (!psLatfile->good()) {         // Local file not found
     filepath = fDataDir + "/" + filename;
-    psLatfile->open(filepath);			// Try data directory
+    char str[1024];
```

Comment:
In this section, can you use `G4String` rather than <stdlib.h>?  G4String is equivalent to std::string, so you can use the various search functions etc.